processing/src/main/java/org/apache/druid/query/expression/TimestampCeilExprMacro.java

```diff
@@ -93,6 +95,13 @@ public Expr visit(Shuttle shuttle)
       return shuttle.visit(new TimestampCeilExpr(newArgs));
     }
 
+    @Nullable
+    @Override
+    public ExprType getOutputType(InputBindingTypes inputTypes)
+    {
+      return ExprType.LONG;
```

Comment:
The eval function returns `ExprEval.of(null)` if the value is a numeric null. It looks like in that case the output type should be ExprType.STRING

Reply:
Hmm, so numeric null handling should be done currently be done using `ExprEval.isNumericNull`, which will return true even if the type is a null string from a `StringExprEval`. So i think 'string' is technically correct, but I don't think spiritually or functionally quite true, nor that useful. We probably should modify these `ExprEval.of(null)` to make type correct `ExprEval` to make `eval` match exactly what `getOutputType` says, but I don't know if this PR is the correct place to do that.

Reply:
Thanks for the explanation. I agree that we shouldn't try to address that in this PR